DashAI/back/database/models.py

```diff
@@ -1,73 +1,61 @@
-from sqlalchemy import Column, Integer, String
+from typing import List
 
-from DashAI.back.database import db
+from sqlalchemy import JSON, Boolean, Enum, ForeignKey, String
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column, relationship
 
+from DashAI.back.models.enums.states import State
 
-class Dataset(db.Base):
+
+class Base(DeclarativeBase):
+    pass
+
+
+class Dataset(Base):
     __tablename__ = "dataset"
     """
-    Class to store all the information about a dataset.
+    Table to store all the information about a dataset.
     """
-    id = Column(Integer, primary_key=True)
-    name = Column(String, unique=True)
-    task_name = Column(String)
-    path = Column(String)
+    id: Mapped[int] = mapped_column(primary_key=True)
+    name: Mapped[str] = mapped_column(String, unique=True, nullable=False)
+    task_name: Mapped[str] = mapped_column(String, nullable=False)
+    file_path: Mapped[str] = mapped_column(String, nullable=False)
+    experiments: Mapped[List["Experiment"]] = relationship()
 
-    def __init__(self, dataset_name, task_name, path):
-        self.name = dataset_name
-        self.task_name = task_name
-        self.path = path
 
+class Experiment(Base):
+    __tablename__ = "experiment"
+    """
+    Table to store all the information about a model.
+    """
+    id: Mapped[int] = mapped_column(primary_key=True)
+    dataset_id: Mapped[int] = mapped_column(ForeignKey("dataset.id"))
+    task_name: Mapped[str] = mapped_column(String, nullable=False)
+    step: Mapped[Enum] = mapped_column(Enum(State), nullable=False)
+    models: Mapped[List["Model"]] = relationship()
 
-# TODO These tables need to be rethinked.
-# class Experiment(db.Base):
-#     __tablename__ = "experiment"
-#     """
-#     Class to store all the information about the experiment
-#     the user do.
-#     """
-#     id = Column(Integer, primary_key=True)
-#     dataset = Column(JSON)
-#     task_filepath = Column(String)
-#     executions = relationship("Execution", back_populates="experiment")
-
-#     def __init__(self, dataset, task_filepath=""):
-#         self.dataset = dataset
-#         self.task_filepath = task_filepath
-
-#     def get_results(self):
-#         results = {}
-#         for execution in self.executions:
-#             results[execution.id] = {
-#                 "train": execution.train_results,
-#                 "test": execution.test_results,
-#             }
-#         return results
 
+class Model(Base):
```

Comment:
Maybe it would be a good idea to change the name to `ModelConfig` or `ModelInstance`, so that it is understood that this is a specific configuration of a particular model and not a table with the specification of each model.

Reply:
Yes! I forgot to change that, thanks